### PR TITLE
Mention  `Study.stop`  as a criteria to stop creating trials in document

### PR DESCRIPTION
**Pull request by takoika**
## Motivation

This PR address #3486. 

## Description of the changes

Based on current description, I have newly mentioned `Study.stop` as one of condition to stop creating new trials to docstring.

### CODE REVIEW

**Comment by takoika** [comment]
I have created the PR in order to satisfy requirements with minimum changes. 

I think that describing conditions to stop creating new trials clearly is better. How about like below?
```
The number of trials for each process. None represents no limit in terms of the number of trials. The Study continue to create trials until the number of trial reaches n_trials, timeout period elapse, Study.stop() is called or a termination signal is received such as SIGTERM or Ctrl+C.
```

**Review by keisuke-umezawa — APPROVED** [review]
LGTM! Thank you for your 1st contribution!

**Review by HideakiImamura — COMMENTED** [review]
> I think that describing conditions to stop creating new trials clearly is better. How about like below?

Thank you for the careful check. :+1: for this change. Could you change the document like that?

**Comment by takoika** [comment]
> Thank you for the careful check. 👍 for this change. Could you change the document like that?

Sure! I will change the MR following my proposal. Please review again after updating!

File: optuna/study/study.py [hunk 1]
```diff
@@ -338,19 +338,19 @@ def objective(trial):
             func:
                 A callable that implements objective function.
             n_trials:
-                The number of trials for each process. If this argument is set to :obj:`None`,
-                there is no limitation on the number of trials. If ``timeout`` is also set to
-                :obj:`None`, the study continues to create trials until it receives a termination
-                signal such as Ctrl+C or SIGTERM.
+                The number of trials for each process. :obj:`None` represents no limit in terms of
+                the number of trials. The Study continue to create trials until the number of
```
Comment (nzw0301) [inline optuna/study/study.py:342]:
```suggestion
                the number of trials. The study continues to create trials until the number of
```

File: optuna/study/study.py [hunk 2]
```diff
@@ -338,19 +338,19 @@ def objective(trial):
...
+                trial reaches `n_trials`, `timeout` period elapse, :func:`~optuna.study.Study.stop`
+                is called or a termination signal such as SIGTERM or Ctrl+C is received.
```
Comment (nzw0301) [inline optuna/study/study.py:344]:
```suggestion
                is called, or a termination signal such as SIGTERM or Ctrl+C is received.
```

File: optuna/study/study.py [hunk 3]
```diff
@@ -338,19 +338,19 @@ def objective(trial):
             func:
                 A callable that implements objective function.
             n_trials:
-                The number of trials for each process. If this argument is set to :obj:`None`,
-                there is no limitation on the number of trials. If ``timeout`` is also set to
-                :obj:`None`, the study continues to create trials until it receives a termination
-                signal such as Ctrl+C or SIGTERM.
+                The number of trials for each process. :obj:`None` represents no limit in terms of
+                the number of trials. The Study continue to create trials until the number of
+                trial reaches ``n_trials``, ``timeout`` period elapse, :func:`~optuna.study.Study.stop`
```
Comment (g-votte) [inline optuna/study/study.py:343]:
```suggestion
                trial reaches ``n_trials``, ``timeout`` period elapses, :func:`~optuna.study.Study.stop`
```

**Comment by HideakiImamura** [comment]
@takoika Thanks for the update. Could you fix the length of the line to pass the CI?

File: optuna/study/study.py [hunk 4]
```diff
@@ -338,19 +338,21 @@ def objective(trial):
             func:
                 A callable that implements objective function.
             n_trials:
-                The number of trials for each process. If this argument is set to :obj:`None`,
-                there is no limitation on the number of trials. If ``timeout`` is also set to
-                :obj:`None`, the study continues to create trials until it receives a termination
-                signal such as Ctrl+C or SIGTERM.
+                The number of trials for each process. :obj:`None` represents no limit in terms of
+                the number of trials. The study continues to create trials until the number of
+                trial reaches ``n_trials``, ``timeout`` period elapses,
```
Comment (g-votte) [inline optuna/study/study.py:343]:
```suggestion
                trials reaches ``n_trials``, ``timeout`` period elapses,
```

File: optuna/study/study.py [hunk 5]
```diff
@@ -338,19 +338,21 @@ def objective(trial):
...
                     :class:`optuna.study.MaxTrialsCallback` can ensure how many times trials
                     will be performed across all processes.
             timeout:
-                Stop study after the given number of second(s). If this argument is set to
-                :obj:`None`, the study is executed without time limitation. If :obj:`n_trials` is
-                also set to :obj:`None`, the study continues to create trials until it receives a
-                termination signal such as Ctrl+C or SIGTERM.
+                Stop study after the given number of second(s). :obj:`None` represents no limit in
+                terms of elapsed time. The Study continue to create trials until the number of
+                trial reaches ``n_trials``, ``timeout`` period elapse,
```
Comment (g-votte) [inline optuna/study/study.py:353]:
```suggestion
                trials reaches ``n_trials``, ``timeout`` period elapses,
```

**Review by g-votte — CHANGES_REQUESTED** [review]
Sorry for frequent suggestions, but there still exist minor English errors.

**Review by g-votte — APPROVED** [review]
LGTM

**Pull request merged**